pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -112,6 +112,31 @@ func (m DIDCommMsgMap) ID() string {
 	return res
 }
 
+// Encode converts struct to message
+func (m DIDCommMsgMap) Encode(v interface{}) error {
```

Comment:
This would make more sense as a type of constructor rather than a member function.

Reply:
Sure! It will. Also `NewDIDCommMsgMap` will be renamed to `ParseDIDCommMsg`

Reply:
Done!